[Docs] New Drivers and ORMs Section (#10878)
yugabyte/yugabyte-db: docs/content/latest/drivers-orms/java/hibernate.md

[Hibernate](https://hibernate.org/orm/) is a popular ORM provider for Java applications which is widely used by Java Developers for Database access. YugabyteDB provides full support for Hiberante ORM and also can be easily used in any environment supporting Java Persistence API (JPA) including Java SE applications, and Java EE application servers connecting to YugabyteDB cluster.

### Step 1: Add the Hibernate ORM Dependency

If you are using [Maven](https://maven.apache.org/guides/development/guide-building-maven.html), add the following to your `pom.xml` of your project.

```xml
<dependency>
    <groupId>org.hibernate</groupId>
    <artifactId>hibernate-core</artifactId>
    <version>5.4.19.Final</version>
</dependency>

<dependency>
    <groupId>org.hibernate</groupId>
    <artifactId>hibernate-annotations</artifactId>
    <version>3.5.6-Final</version>
</dependency>
```

If you are using [Gradle](https://docs.gradle.org/current/samples/sample_building_java_applications.html), add the following dependencies to your `build.gradle` file:

```java
implementation 'org.hibernate:hibernate-core:5.4.19.Final'
implementation 'org.hibernate:hibernate-annotations:3.5.6-Final'
```

> Note: Hibernate ORM can be used with both [YugabyteDB JDBC driver](./yugabyte-jdbc) and [Postgres JDBC Driver](./postgres-jdbc).

Author: fixed.

        <property name="hibernate.dialect">org.hibernate.dialect.PostgreSQLDialect</property>
        <property name="hibernate.connection.driver_class">com.yugabytedb.Driver</property>
        <property name="hibernate.connection.url">jdbc:yugabytedb://localhost:5433/yugabyte</property>
        <property name="hibernate.connection.username">yugabyte</property>
        <property name="hibernate.connection.password"></property>
        <property name="hibernate.hbm2ddl.auto">update</property>
        <property name="show_sql">true</property>
        <property name="generate-ddl">true</property>
        <property name="hibernate.ddl-auto">generate</property>
        <property name="hibernate.connection.isolation">8</property>
        <property name="hibernate.current_session_context_class">thread</property>
        <property name="javax.persistence.create-database-schemas">true</property>
    </session-factory>
</hibernate-configuration>
```

The Hibernate configuration file provides the generic set of properties that are required for configuring the Hibernate ORM for YugabyteDB.

| Hibernate Params | Description | Default |
| :---------- | :---------- | :------ |

> | hibernate.dialect  | Dialect is used by hibernate to generate SQL optimized for a particular relational database | org.hibernate.dialect.PostgreSQLDialect

Reviewer: ```suggestion
| hibernate.dialect  | Dialect to use to generate SQL optimized for a particular database | org.hibernate.dialect.PostgreSQLDialect
```

> | hibernate.connection.driver_class | provides the JDBC Driver name  | com.yugabytedb.Driver

Reviewer: ```suggestion
| hibernate.connection.driver_class | JDBC Driver name  | com.yugabytedb.Driver
```

| hibernate.connection.url | JDBC Connection URL | jdbc:yugabytedb://localhost:5433/yugabyte
| hibernate.connection.username | Username | yugabyte
| hibernate.connection.password | Password | yugabyte

> | hibernate.hbm2ddl.auto | Controls the behavior for Automatic schema generation | none

Reviewer: ```suggestion
| hibernate.hbm2ddl.auto | Behaviour for automatic schema generation | none
```

Hibernate provides an [exhaustive list of properties](https://docs.jboss.org/hibernate/orm/5.6/userguide/html_single/Hibernate_User_Guide.html#configurations-general) to configure the different features supported by the ORM. Additional details can be obtained by referring to the [Hibernate documentation](https://hibernate.org/orm/documentation/5.6/).

### Step 5: Adding the Object relational mapping

Along with properties for configuring the Hibernate ORM, `hibernate.cfg.xml` is also used for specifying the Domain objects mapping using `<mapping>` tags.

Add a mapping for `Employee` object in `hibernate.cfg.xml`

```xml
<?xml version="1.0" encoding="utf-8"?>
<!DOCTYPE hibernate-configuration SYSTEM
        "http://www.hibernate.org/dtd/hibernate-configuration-3.0.dtd">

<hibernate-configuration>
    <session-factory>
        ...
        <mapping class="com.yugabyte.hibernatedemo.model.Employee"/>
    </session-factory>
</hibernate-configuration>
```

### Step 6: Query the YugabyteDB Cluster using Hibernate ORM

> Create a new Java class called `QuickStartOrmApp.java` in the base package directory of your project. Copy the sample code below in order to query the table contents from the java client using Hibernate ORM. Ensure you replace the connection string `yburl` with the credentials of your cluster and SSL certificates if required.

Reviewer: ```suggestion
Create a new Java class called `QuickStartOrmApp.java` in the base package directory of your project. Copy the following sample code to query the table contents from the Java client using Hibernate ORM. Ensure you replace the parameters in the connection string `yburl` with the cluster credentials and SSL certificate, if required.
```
